### PR TITLE
Support Decimal type

### PR DESCRIPTION
**Pull request by maxim432**
Adds first-class Python `decimal.Decimal` support via rust_decimal, enabling 
precise decimal arithmetic without float precision loss.

Changes:
- New `decimal.rs` with Starlark wrapper of rust_decimal
- Bidirectional Python↔Starlark conversion preserving Decimal precision in `lib.rs`
- 13 tests covering operations, edge cases, and error handling
- README documentation with usage example

Intentionally rejects float→Decimal coercion to prevent silent precision loss.

### CODE REVIEW

**Comment by JakobDegen** [comment]
@inducer tbh I don't really know. I don't think we have much of a policy around this. That being said, it's not obvious to me what you get from fully upstreaming vs potentially releasing as a standalone library, so I think my default answer here might be to maybe not. As I said though, I'm not sure and don't feel strongly

File: src/decimal.rs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+use starlark::starlark_simple_value;
+use starlark::values::{
+    Heap, NoSerialize, ProvidesStaticType, StarlarkValue, Value, ValueError, ValueLike,
+};
+use starlark_derive::{starlark_module, starlark_value};
+
+#[derive(Debug, Clone, ProvidesStaticType, NoSerialize, Allocative)]
+pub struct DecimalValue {
+    #[allocative(skip)]
+    pub(crate) value: Decimal,
```
[the remark this answers is not in the document]
Comment (maxim432) [reply src/decimal.rs]: Good points about semantic differences - I've documented them in README. Not sure if there is much more that could be done. If you think 'Decimal' is misleading, I'm open to call it 'FixedDecimal' or similar to emphasize the difference.

[the remark this answers is not in the document]
Comment (inducer) [reply src/decimal.rs]: How about `RustDecimal`? On the one hand, the identifier "sticks out like a sore thumb", on the other it's at least precise/honest.

File: src/decimal.rs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+/// Convert DecimalValue to Python Decimal object
+pub fn decimal_to_python(decimal_value: &DecimalValue) -> PyResult<Py<PyAny>> {
+    let decimal_str = decimal_value.value.to_string();
+    Python::attach(|py| {
+        let decimal_module = py.import("decimal")?;
+        let decimal_type = decimal_module.getattr("Decimal")?;
+        let py_decimal = decimal_type.call1((decimal_str,))?;
+        Ok(py_decimal.into())
+    })
```
Comment (inducer) [inline src/decimal.rs]: This should probably try to match the precision set on the Rust side, otherwise any operations on the Python side will have surprising rounding behavior.

**Review by inducer — COMMENTED** [review]
Thanks! I think I see a way towards merging this, but it'll need a bit more work, see below.

File: src/decimal.rs [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+}
+
+pub fn alloc_decimal<'v>(heap: &'v Heap, decimal: Decimal) -> Value<'v> {
+    heap.alloc(DecimalValue { value: decimal })
+}
+
+#[starlark_module]
+pub fn decimal_module(builder: &mut GlobalsBuilder) {
+    /// Construct a decimal value from a string, int, or existing decimal.
+    fn Decimal<'v>(#[starlark(require = pos)] value: Value<'v>, heap: &'v Heap) -> starlark::Result<Value<'v>> {
```
[the remark this answers is not in the document]
Comment (maxim432) [reply src/decimal.rs]: exposed scale and round_dp methods
